crates/datasources/src/mongodb/mod.rs

```diff
@@ -332,6 +333,18 @@ impl TableProvider for MongoDbTableProvider {
             self.estimated_count,
         )))
     }
+
+    async fn insert_into(
+        &self,
+        _state: &SessionState,
+        input: Arc<dyn ExecutionPlan>,
+        _overwrite: bool,
```

Comment:
I am not sure if we do support overwrite. Let's throw an error here if `overwrite` is `true`.